### PR TITLE
feat: reset token and mailer

### PR DESCRIPTION
**Pull request by r-dmatos**
Relates to  #108

## Proposed Changes
  -  Reset token schema and create and update method implemented.
  -  Mailer module was used to send an email with a link to reset the user password .
  - A 1 minute timer was implemented to the reset password flow to prevent extra requests from the same user ( there might be a better way to do this ) 
  - Env files updated with mailer module options

### CODE REVIEW

**Comment by nunocaseiro** [comment]
@all-contributors please add @r-dmatos for code and doc

File: backend/src/infrastructure/database/mongoose.module.ts [hunk 1]
```diff
@@ -20,6 +23,9 @@ export const mongooseBoardUserModule = MongooseModule.forFeature([
 export const mongooseUserModule = MongooseModule.forFeature([
   { name: User.name, schema: UserSchema },
 ]);
+export const mongooseResetModule = MongooseModule.forFeature([
```
Comment (rpvsilva) [inline backend/src/infrastructure/database/mongoose.module.ts:26]:
```suggestion
]);

export const mongooseResetModule = MongooseModule.forFeature([
```

File: backend/src/modules/auth/controller/auth.controller.ts [hunk 2]
```diff
@@ -88,6 +91,12 @@ export default class AuthController {
     return this.getUserApp.getByEmail(email).then((user) => !!user);
   }
 
+  @Post('recoverPassword')
+  async forgot(@Body() emailDto: EmailParam) {
+    const { email } = emailDto;
+    return this.createResetTokenAuthApp.create(email);
+  }
```
Comment (rpvsilva) [inline backend/src/modules/auth/controller/auth.controller.ts:98]:
```suggestion
  forgot(@Body() { email }: EmailParam) {
    return this.createResetTokenAuthApp.create(email);
  }
```

File: backend/src/modules/auth/services/create-reset-token.auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  implements CreateResetTokenAuthService
+{
+  constructor(
+    @InjectModel(ResetPassword.name)
+    private resetModel: Model<ResetPasswordDocument>,
+    private mailerService: MailerService,
+    private configService: ConfigService,
+  ) {}
+
+  private frontendUrl = this.configService.get<string>('frontend.url');
```
Comment (rpvsilva) [inline backend/src/modules/auth/services/create-reset-token.auth.service.ts:23]: set this variable before the constructor and then set it inside it

File: backend/src/modules/auth/services/create-reset-token.auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      to: emailAddress,
+      subject: 'You requested a password reset',
+      html: `Trouble signing in?
+      Resetting your password is easy.
+      
+      Just click <a href ="${url}"> here </a> to reset your password. We’ll have you up and running in no time.
+      
+      If you did not make this request then please ignore this email.`,
+    });
+    return { message: 'please check your email' };
```
Comment (rpvsilva) [inline backend/src/modules/auth/services/create-reset-token.auth.service.ts:37]: Define a constant with this message

File: backend/src/modules/auth/services/create-reset-token.auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+    try {
+      const passwordModel = await this.resetModel.findOne({ emailAddress });
+      if (!passwordModel) throw Error(INSERT_FAILED);
+      this.tokenValidator(passwordModel?.updatedAt);
+
+      const { token } = await this.tokenGenerator(emailAddress, session);
+      if (token) {
+        await session.commitTransaction();
+        return await this.emailBody(token, emailAddress);
```
Comment (rpvsilva) [inline backend/src/modules/auth/services/create-reset-token.auth.service.ts:77]: You should commit the transaction after the `emailBody`, so set a constant and await for the method and then commit then return

**Review by rpvsilva — CHANGES_REQUESTED** [review]

**Review by nunocaseiro — APPROVED** [review]
LGTM 🚀

**Review by rpvsilva — DISMISSED** [review]
LGTM 🚀

**Review by nunocaseiro — APPROVED** [review]

**Pull request merged**